pubtools/_quay/tag_docker.py

```diff
@@ -722,11 +724,30 @@ def merge_manifest_lists_sign_images(
         ) == sorted(
             json.loads(raw_src_manifest)["manifests"], key=lambda manifest: manifest["digest"]
         ):
+            ml_to_sign = cast(ManifestList, raw_src_manifest)
             self.quay_client.upload_manifest(raw_src_manifest, dest_image, raw=True)
         else:
+            ml_to_sign = cast(ManifestList, new_manifest_list)
             self.quay_client.upload_manifest(new_manifest_list, dest_image)
 
         if push_item.claims_signing_key:
+            # for cosign sign also manifest list
+            to_sign_entries_internal.append(
+                SignEntry(
+                    repo=list(push_item.repos.keys())[0],
+                    reference="quay.io/"
+                    + self.target_settings["quay_namespace"]
+                    + "/"
+                    + internal_repo
+                    + ":"
+                    + tag,
+                    digest="sha256:"
+                    + hashlib.sha256(json.dumps(ml_to_sign).encode("utf-8")).hexdigest(),
```

Comment:
Digest is incorrect.
```suggestion
                    + hashlib.sha256(ml_to_sign.encode("utf-8")).hexdigest(),
```